unitex/src/fr/umlv/unitex/frames/FindAndReplaceDialog.java

```diff
@@ -0,0 +1,633 @@
+    final JPanel spacer8 = new JPanel();
+    gbc = new GridBagConstraints();
+    gbc.gridx = 0;
+    gbc.gridy = 1;
+    gbc.fill = GridBagConstraints.HORIZONTAL;
+    panel1.add(spacer8, gbc);
+    final JPanel spacer9 = new JPanel();
+    gbc = new GridBagConstraints();
+    gbc.gridx = 3;
+    gbc.gridy = 13;
+    gbc.fill = GridBagConstraints.VERTICAL;
+    panel1.add(spacer9, gbc);
+    final JPanel spacer10 = new JPanel();
+    gbc = new GridBagConstraints();
+    gbc.gridx = 5;
+    gbc.gridy = 0;
+    gbc.fill = GridBagConstraints.VERTICAL;
+    panel1.add(spacer10, gbc);
+    final JPanel spacer11 = new JPanel();
+    gbc = new GridBagConstraints();
+    gbc.gridx = 6;
+    gbc.gridy = 1;
+    gbc.fill = GridBagConstraints.HORIZONTAL;
+    panel1.add(spacer11, gbc);
+    ignoreCommentBoxesCheckBox = new JCheckBox();
+    ignoreCommentBoxesCheckBox.setText("Ignore comment boxes");
+    ignoreCommentBoxesCheckBox.setSelected(true);
+    gbc = new GridBagConstraints();
+    gbc.gridx = 5;
+    gbc.gridy = 8;
+    gbc.anchor = GridBagConstraints.WEST;
+    panel1.add(ignoreCommentBoxesCheckBox, gbc);
+    useRegularExpressionsCheckBox = new JCheckBox();
+    useRegularExpressionsCheckBox.setText("Use regular expressions");
+    gbc = new GridBagConstraints();
+    gbc.gridx = 5;
+    gbc.gridy = 7;
+    gbc.anchor = GridBagConstraints.WEST;
+    panel1.add(useRegularExpressionsCheckBox, gbc);
+    final JPanel spacer12 = new JPanel();
+    gbc = new GridBagConstraints();
+    gbc.gridx = 5;
+    gbc.gridy = 9;
+    gbc.fill = GridBagConstraints.VERTICAL;
+    panel1.add(spacer12, gbc);
+    findTextField = new JTextField("");
+    findTextField.setEditable(true);
+    gbc = new GridBagConstraints();
+    gbc.gridx = 3;
+    gbc.gridy = 1;
+    gbc.gridwidth = 3;
+    gbc.anchor = GridBagConstraints.WEST;
+    gbc.fill = GridBagConstraints.HORIZONTAL;
+    panel1.add(findTextField, gbc);
+    replaceTextField = new JTextField("");
+    replaceTextField.setEditable(true);
+    gbc = new GridBagConstraints();
+    gbc.gridx = 3;
+    gbc.gridy = 3;
+    gbc.gridwidth = 3;
+    gbc.anchor = GridBagConstraints.WEST;
+    gbc.fill = GridBagConstraints.HORIZONTAL;
+    panel1.add(replaceTextField, gbc);
+    return panel1;
+  }
+
+  private void fillComboBox() {
+    ComboBoxToolTipRenderer renderer = new ComboBoxToolTipRenderer();
+    ArrayList<String> tooltips = new ArrayList<String>();
+    DefaultComboBoxModel<GraphFrame> model = new DefaultComboBoxModel<GraphFrame>();
+    model.addElement(new defaultGraphFrame());
+    tooltips.add(graphDefaultText);
+    for (GraphFrame f : graphFrames) {
+      model.addElement(f);
+      if (f.getGraph() == null) {
+        tooltips.add(f.toString());
+      } else {
+        tooltips.add(f.getGraph().getPath());
+      }
+    }
+    renderer.setTooltips(tooltips);
+    graphComboBox.setRenderer(renderer);
+    graphComboBox.setModel(model);
+  }
+
+  private void onClose() {
+    dispose();
```

Comment:
I've just checked that usually in UnitexFrames closing action for buttons is defined right in the `actionPerformed(ActionEvent e){  }` body. For example [here](https://github.com/UnitexGramLab/gramlab-ide/blob/master/unitex/src/fr/umlv/unitex/frames/ProcessInfoFrame.java) lines 139-141